crates/sui-core/src/transaction_driver/message_types.rs

```diff
@@ -47,6 +48,21 @@ pub enum SubmitTxResponse {
     },
 }
 
+impl fmt::Debug for SubmitTxResponse {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Submitted { consensus_position } => f
+                .debug_struct("Submitted")
+                .field("consensus_position", consensus_position)
+                .finish(),
+            Self::Executed { effects_digest, .. } => f
+                .debug_struct("Executed")
+                .field("effects_digest", effects_digest)
```

Comment:
Deliberately excluding the `ExecutedData`  from logs as I don't think they'll give much value via overlogging.